src/components/plot/user-controls-grid.tsx

```diff
@@ -33,27 +37,123 @@ const UserControlsGrid = ({
+      }
+    }
+
+    //Validate local controls with same uid. Display error in case of same uid(s)
+    const uniqueUid: string[] = [];
+    const dupUid: string[] = [];
+    const controlObject = localControlData.reduce((result, control) => {
+      if (!uniqueUid?.includes(control.uid)) {
+        uniqueUid.push(control.uid)
+      } else {
+        if (!dupUid?.includes(control.uid)) {
+          dupUid.push(control.uid);
+        }
+      }
+      result[control.uid] = control;
+      return result;
+    }, {});
+    if (dupUid?.length > 0) {
+      dupUid.map((id) => alertFunctions.addAlert(`Multiple local controls with the same UID '${id}' were detected`, ChaiseAlertType.ERROR));
+    }
+    userControlData.userControlConfig = Object.values(controlObject);
+    setLocalControlData(Object.values(controlObject));
+
+    // Validate controls for not having neither of the request_info.data nor the request_info.url_pattern and display error
+    const invalidControlData: string[] = [];
+    const validatedUserControls = userControlData.userControlConfig?.map((control: UserControlConfig) => {
+      let isControlValid = true;
+      if (!((control.request_info?.data && control.request_info?.data?.length > 0) || control.request_info?.url_pattern)) {
+        isControlValid = false;
+        if (!alertFunctions.alerts.some(
+          (alert) => alert.message.includes(`Unable to display the local control with Uid '${control.uid}'`))) {
+          invalidControlData.push(control.uid);
+        }
+      }
+      return ({ ...control, visible: isControlValid });
+    }).filter((control) => control.visible || control?.visible == undefined);
+    if (invalidControlData?.length > 0) {
+      alertFunctions.addAlert(`Unable to display the local control with Uid(s): ${invalidControlData.join(', ')} because neither the request_info.data nor the request_info.url_pattern were found`, ChaiseAlertType.WARNING);
+    }
+    setLocalControlData(validatedUserControls);
+
+    if (userControlData?.gridConfig?.layouts && Object.values(userControlData.gridConfig?.layouts)?.length > 0) {
       const mappedLayoutValues = Object.values(userControlData.gridConfig?.layouts)?.map((resLayout: any) => (
         resLayout.map((item: LayoutConfig) => convertKeysSnakeToCamel(({
           // i defines the item on which the given layout will be applied
           i: item?.source_uid,
           ...item,
         })))
       ));
-
-      setLayout(Object.fromEntries(Object.entries(userControlData.gridConfig?.layouts).map(([key]: any, index) => [key, mappedLayoutValues[index]])));
+      let tempLayout;
+      tempLayout = Object.fromEntries(Object.entries(userControlData.gridConfig?.layouts).map(([key]: any, index) => [key, mappedLayoutValues[index]]));
+      const invalidLayoutUid: string[] = [];
+
+      // Check if the layout parameter has all the necessary properties such as source_uid, h, w, x, and y. If any of these properties are missing, display a warning.
```

Comment:
This is a 4th function we can factor out to `plot-utils` since we are validating the layout the same way in both places.